animal-sniffer/src/main/java/org/codehaus/mojo/animal_sniffer/Clazz.java

```diff
@@ -59,20 +59,56 @@ public final class Clazz
      */
     private final String[] superInterfaces;
 
+    /**
+     * Internal constructor which just assigns all fields, without performing any defensive copying
+     * or similar.
+     *
+     * @param dummy Unused; only needed to avoid constructor signature conflicts
+     */
+    private Clazz( String name, Set<String> signatures, String superClass, String[] superInterfaces, Void dummy ) {
+        this.name = name;
+        this.signatures = signatures;
+        this.superClass = superClass;
+        this.superInterfaces = superInterfaces;
+    }
+
+    /**
+     * Creates a new class signature, with an initially empty set of member signatures.
+     *
+     * @param name            the name of the class.
+     * @param superClass      the superclass.
+     * @param superInterfaces the interfaces implemented by the class; a copy of this array is created.
+     */
+    public Clazz( String name, String superClass, String[] superInterfaces )
```

Comment:
Created this separate constructor (and the `private` one above) because the usage in `SignatureBuilder` just provided an empty set and also with the code changes here a defensive copy of that would have been made. So it felt less error-prone to me to have a separate constructor without signatures `Set` parameter.